cli/tests/transfer.rs

```diff
@@ -340,3 +356,57 @@ fn test_transfer_multisession_signing() {
     server.close().unwrap();
     remove_dir_all(ledger_path).unwrap();
 }
+
+#[test]
+fn test_transfer_all() {
+    let TestValidator {
+        server,
+        leader_data,
+        alice: mint_keypair,
+        ledger_path,
+        ..
+    } = TestValidator::run_with_options(TestValidatorOptions {
+        fees: 1,
+        bootstrap_validator_lamports: 42_000,
+        ..TestValidatorOptions::default()
+    });
+
+    let (sender, receiver) = channel();
+    run_local_faucet(mint_keypair, sender, None);
+    let faucet_addr = receiver.recv().unwrap();
+
+    let rpc_client = RpcClient::new_socket(leader_data.rpc);
+
+    let default_signer = Keypair::new();
+
+    let mut config = CliConfig::default();
+    config.json_rpc_url = format!("http://{}:{}", leader_data.rpc.ip(), leader_data.rpc.port());
+    config.signers = vec![&default_signer];
+
+    let sender_pubkey = config.signers[0].pubkey();
+    let recipient_pubkey = Pubkey::new(&[1u8; 32]);
+
+    request_and_confirm_airdrop(&rpc_client, &faucet_addr, &sender_pubkey, 50_000, &config)
```

Comment:
Why do this airdrop (adds ~12s to CI)? As opposed to using mint_keypair?

Reply:
Copy-pasta :-/... which maybe means we can cut 12s off each of a whole bunch of tests. I'll take a look!

Reply:
I was hoping for an easy CI win here, but it turns out that if I remove that airdrop, the fees get wonky because it pulls the fee calculator from block 0 (commitment max), and the `FeeCalculator::lamports_per_signature` for block 0 = 0. A sleep of ~12s fixes the problem 😆 .
For this test, we could assume 0 fees on the single transaction, nbd. But it would be flaky in other tests where we have to guess when fees will be 0 vs non-zero, based on when slot 32 will happen.

I'm going to leave this as-is for now, and fix as part of a broader effort to cut out all the 12s waits in all these integration tests.

Reply:
FWIW, the fee rate is stable on slot 2. It's 0 in genesis, copied from genesis to Slot1, updated in Slot1 then copied to Slot2.